### PR TITLE
Task persistance a.k.a. Code for saving task manager state

### LINKED ISSUE #1339: Check code for saving task manager state
Check the code for dumping task manager state & task keeper state, do any necessary refactoring & changes.

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #1339

### CODE REVIEW

File: golem/task/taskmanager.py [hunk 1]
```diff
@@ -208,27 +209,28 @@ def remove_dump(self, task_id: str):
...
                 try:
                     task, state = pickle.load(f)
                     self.tasks[task.header.task_id] = task
                     self.tasks_states[task.header.task_id] = state
+
+                    dispatcher.send(
+                        signal='golem.taskmanager',
+                        event='task_status_updated',
+                        task_id=task.header.task_id
+                    )
```
Comment (jiivan) [inline golem/task/taskmanager.py]: I think that moving `dispatcher.send` inside `try` and `with` block is a mistake. File should be closed before dispatching, and try block handles unpickling related errors.

**Review by ggruszczynski — APPROVED** [review]

File: golem/task/taskmanager.py [hunk 2]
```diff
@@ -168,57 +171,85 @@ def start_task(self, task_id):
...
+            logger.info("Task %s added" % task.header.task_id)
             self.notice_task_updated(task.header.task_id)
 
-    def dump_task(self, task_id):
-        logger.debug('DUMP TASK')
+    def _dump_filepath(self, task_id):
+        return self.tasks_dir / ('%s.pickle' % (task_id,))
+
+    def dump_task(self, task_id: str) -> None:
+        logger.debug('DUMP TASK CALLED')
```
Comment (Elfoniok) [inline golem/task/taskmanager.py]: Please add task id to logs

File: golem/task/taskmanager.py [hunk 3]
```diff
@@ -72,32 +70,39 @@ def __init__(self, node_name, node, keys_auth, listen_address="",
...
         self.listen_address = listen_address
         self.listen_port = listen_port
 
-        # FIXME Remove this variable and make task persistance obligatory after it is more tested
-        # Remember to also remove it from init params
         self.task_persistence = task_persistence
 
-        self.tasks_dir = Path(tasks_dir)
+        tasks_dir = Path(tasks_dir)
+        self.tasks_dir = tasks_dir / "tmanager"
```
Comment (Elfoniok) [inline golem/task/taskmanager.py:83]: Can you get this dir from config?

**Review by Elfoniok — CHANGES_REQUESTED** [review]
Please revert changes that delete task.state updates. There were done very poorly (only some random set instructions were removed) moreover, i think its bad idea to remove state from task. Actually this is task state. Its true tat it looks like those two (task_states dict and state in task) should be unified. But this should be done in separate PR. Probably it would be better to remove dict itself.

**Review by Elfoniok — APPROVED** [review]

File: golem/task/taskmanager.py (hunk 3 above)
Comment (etam) [reply golem/task/taskmanager.py:83]: Shouldn't this be managed by DirManager? It already does something similar for other directories.

**Pull request merged**